[14.0][ADD] mrp_subcontracting_picking

ivs-cetmix opened the pull request:
Adding the mrp_subcontracting_picking module

ivs-cetmix committed:
[ADD] mrp_subcontracting_picking

ivs-cetmix committed:
[ADD] mrp_subcontracting_picking

elvise commented:
@isserver1 please double check runboat :)

elvise commented:
@isserver1 gentle reminder about the runboat :)

ivs-cetmix commented:
@OCA/manufacturing-maintainers could you check this please?

pedrobaeza commented on mrp_subcontracting_picking/README.rst:
```diff
@@ -0,0 +1,31 @@
+========================================
```
This README has incorrect syntax and it's not complete without description

ivs-cetmix replied:
Thank you for mentioning this! Fixed this

pedrobaeza commented on mrp_subcontracting_picking/readme/DESCRIPTION.rst:
```diff
@@ -0,0 +1,12 @@
+Ooops Subcontracting Picking
```
Please don't put company name nor title here.

ivs-cetmix replied:
Done

pedrobaeza commented on mrp_subcontracting_picking/readme/HISTORY.rst:
```diff
@@ -0,0 +1,4 @@
+14.0.1.0.0
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
No need of this file, but if put, the underlining should be of the same length of the contents

ivs-cetmix replied:
Done

pedrobaeza commented on mrp_subcontracting_picking/readme/DESCRIPTION.rst:
```diff
@@ -0,0 +1,12 @@
+Subcontracting Picking
```
This title should go out.

pedrobaeza reviewed (commented):
Reading the description, I consider this module is transfiguring somehow the standard of Odoo, as the idea is that you don't do 2 steps: manufacture and receipt the subcontracted material. You do it both things through the reception picking. That's why the are put as ready in the state.

geomer198 committed:
[FIX] Model purchase.order : changed _create_pucking func and tests.

geomer198 committed:
[FIX] Model purchase.order : changed _create_pucking func and tests.

elvise commented:
@isserver1 can you rebase ?

elvise commented:
@geomer198 can you rebase ?

geomer198 committed:
Merge branch '14.0-add_mrp_subcontracting_picking' of https://github.com/cetmix/oca-manufacture into 14.0-add_mrp_subcontracting_picking

ivs-cetmix committed:
Merge branch 'OCA:14.0' into 14.0-add_mrp_subcontracting_picking

geomer198 committed:
[FIX] Returned to default behavior and add new functions.

geomer198 committed:
Merge branch '14.0-add_mrp_subcontracting_picking' of https://github.com/cetmix/oca-manufacture into 14.0-add_mrp_subcontracting_picking

ivs-cetmix commented on mrp_subcontracting_picking/models/mrp_production.py:
```diff
@@ -0,0 +1,27 @@
+            ).filtered(lambda p: p.state == "subcontracted")
+            if not pickings:
+                continue
+            for picking in pickings:
```
Why do we need an iteration here?
[action_assign](https://github.com/odoo/odoo/blob/1761d1de485382a5c5dc2d5f421c315ce34e1a3e/addons/stock/models/stock_picking.py#L750) and [action_confirm](https://github.com/odoo/odoo/blob/1761d1de485382a5c5dc2d5f421c315ce34e1a3e/addons/stock/models/stock_picking.py#L738) both work with recordsets

geomer198 replied:
Resolved

ivs-cetmix commented on mrp_subcontracting_picking/models/mrp_production.py:
```diff
@@ -0,0 +1,27 @@
+            pickings = prod.move_finished_ids.mapped(
+                "move_dest_ids.picking_id"
+            ).filtered(lambda p: p.state == "subcontracted")
+            if not pickings:
```
why "if not" if we can use 
`if pickings: `

geomer198 replied:
Resolved

geomer198 committed:
Changes :
[IMP] Added demo data
[FIX] Changed mrp.production model

geomer198 committed:
[FIX] Changed demo data.

geomer198 committed:
[FIX] Changed demo data

Setted active state is True for 'mrp_subcontracting_picking.stock_subcontracting_location' record.

geomer198 committed:
[FIX] Changed demo data.
Changed sequence for stock location route.

geomer198 committed:
[FIX] mrp_subcontracting_picking : Changed demo file and purchase order model.
